clients/src/test/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImplTest.java

```diff
@@ -1161,8 +1155,9 @@ public void testNewAssignmentReplacesPreviousOneWaitingOnMetadata() {
         membershipManager.poll(time.milliseconds());
 
         Set<TopicPartition> expectedAssignment = Collections.singleton(new TopicPartition(topicName, 0));
+        HashSet<TopicPartition> expectedSet = new HashSet<>(expectedAssignment);
```

Comment:
When we ```verify(subscriptionState).assignFromSubscribedAwaitingCallback()```, we want to make sure that the assignments are being calculated and propagated properly. Having an expected set of partitions for the assignment makes the test more strict, yet still correct.

Reply:
@brenden20 I believe @philipnee 's point is about why we need the `expectedSet` and can't get away with `expectedAssignment` only? I think we could indeed remove expectedSet here (we do need it in other similar checks to sort the set or add elems, but since this one has a single elem we don't need to have it I expect)

Reply:
@lianetm I see now, I have made the change

Reply:
isn't `Set<TopicPartition> expectedAssignment ` already a set?